normandy/recipes/models.py

```diff
@@ -122,18 +163,52 @@ def update_signature(self):
 
     @transaction.atomic
     def update(self, force=False, **data):
```

Comment:
This doesn't check all revisions, it only checks the latest_revision to make sure that it has changed. But yes, checking by hash is probably way less complicated and more performant. I'll make the change.